pywbem_mock/_baserepository.py

```diff
@@ -0,0 +1,353 @@
+        self._cim_object_type = cim_object_type
+
+    @abstractmethod
+    def exists(self, name):
+        """
+        Test if cim_object defined by name exists in the
+        object store
+
+        Paramsters
+          name(:term:`string` or :class:`~pywbem.CIMInstanceName`):
+            Name by which the object is identified in the object store
+
+        Returns:
+            True if the object exists; False if it does not exist
+        """
+        pass
+
+    @abstractmethod
+    def get(self, name, copy=True):
+        """
+        Return the cim_object defined by name and namespace if it exists.
+
+        Parameters:
+          name(:term:`string` or :class:`~pywbem.CIMInstanceName`):
+            Name by which the object is identified in the object store
+
+          copy: If True, insures that modifying the returned object does not
+            change the data store. The default behavior is True.  If
+            copy is False, the object in the object store is returned but
+            if it is modified by the user, the object in the store may
+            be modified also.
+
+        Returns:
+            Returns the cim_object identified by the name parameter
+
+        Raises:
+            KeyError: Object not in this object store
+        """
+        pass
+
+    @abstractmethod
+    def create(self, name, cim_object):
+        """
+        Adds the cim_object identified by name to the object store.
+
+        Parameters:
+
+          name(:term:`string` or :class:`~pywbem.CIMInstanceName`):
+            Name by which the object is identified in the object store .
+
+          cim_object(cim_object to be added to the object store):
+            The CIM object that will be inserted into the object store.
+
+        Raises:
+            ValueError: If object already exists in the object store.
+        """
+        pass
+
+    @abstractmethod
+    def update(self, name, cim_object):
+        """
+        Replace the object defined by name in the object store with
+        the object defined by cim_object.
+
+        Parameters:
+
+          name(:term:`string` or :class:`~pywbem.CIMInstanceName`):
+            Name by which the object is identified in the object store.
+
+          cim_object(cim_object to be added to the object store):
+
+        Raises:
+            KeyError: If no object with name exists in the object store.
+        """
+        pass
+
+    @abstractmethod
+    def delete(self, name):
+        """
+        Delete the object identified by name from the object store.
+
+        Parameters:
+
+          name(:term:`string` or :class:`~pywbem.CIMInstanceName`):
+            Name by which the object is identified in the object store.
+
+        Raises:
+            KeyError: If there is no object with name in this object store
+        """
+        pass
+
+    @abstractmethod
+    def iter_names(self):
+        """
+        Return iterator to the names of the objects in the object store.
```

Comment:
Should we say that the iteration order is undefined?

Reply:
YES, Added comment. DONE